README.md

```diff
@@ -1,16 +1,248 @@
+organize our code into chunks of reusable functionality that can be easily used
+by other areas of their program.
 
-## SWBAT 2
+Ruby methods are very similar to functions in any other programming language.
+Ruby methods are used to bundle one or more repeatable statements into a single
+unit.
+
+In this lesson we'll introduce methods, distinguish them from data types, and
+cover how to create and execute them in your Ruby program.
+
+## Describe Routines and Procedures for Our Code Using Methods
+
+Methods define a new thing that your program can do. Variables in Ruby store
+data; methods store a new routine or behavior your program can use. Variables
+are like "nouns", a thing, and methods are like "verbs", an action.
+
+For example, imagine needing to say "Hello World!" five times. It could look
+like this:
+
+```ruby
+phrase = "Hello World!"
+puts phrase
+puts phrase
+puts phrase
+puts phrase
+puts phrase
+```
+
+That technically works. You made use of a variable to store the data you wanted
+to output, and then the next five lines output the phrase.
+
+Now imagine later in your program you want to say "Hello World!" five times
+again. We would have to output the value of `phrase` five times.
+
+If we wanted to use a method to contain this actions, it could look like this:
+
+```ruby
+def say_hello_world_five_times
+  phrase = "Hello World!"
+  puts phrase
+  puts phrase
+  puts phrase
+  puts phrase
+  puts phrase
+end
+```
+
+Now, when we use the word `say_hello_world_five_times` in our program, it
+will invoke the method, running the code within the method. So the script above,
+saying hello five times, doing other things, then saying hello five times again
+could be rewritten as this:
+
+```ruby
+def say_hello_world_five_times
+  phrase = "Hello World!"
+  puts phrase
+  puts phrase
+  puts phrase
+  puts phrase
+  puts phrase
+end
+```
+
+That's much cleaner and follows the code principle "don't repeat yourself" or
+DRY. We abstract the action or procedure of putting "Hello World!" five times
+into a method. By defining the method `say_hello_world_five_times` once, we can
+"call" or "invoke" the method as many times as we want in the future. Let's look
+at methods in greater detail.
+
+## Define a Method
+
+Method names should begin with a lowercase letter. If you begin a method name
+with an uppercase letter, Ruby might think that it is a constant and hence can
+parse the call incorrectly.
+
+Suppose we need to define a simple method that outputs a string.
+
+You can define a method in Ruby with the `def` keyword. A method's name can
+begin with any lowercase letter. Here’s how we can do that:
+
+```ruby
+def greeting # Method Signature
+  puts "Hello World" # Method Body
+end # Method Closing
+```
+
+This defines a method. It does not use it -- yet: This is the first step so that
+it can be used later.
+
+Let’s walk through this method definition step by step:
```

Comment:
I feel like there's at least 1-2 hidden learning goals in here that aren't named. Try to flush those out.